### PR TITLE
allow roundrobin algorithm as a choice in passthrough/reencrypt

### PR DESCRIPTION
**Pull request by rajatchopra**
For the passthrough and reencrypt routes, the choice of balancing algorithm was not provided through annotations. Making that possible with this PR. Weights depend on the fact that the balancing algorithm be 'roundrobin'.

### CODE REVIEW

**Comment by rajatchopra** [comment]
[test]

**Comment by rajatchopra** [comment]
@openshift/networking @ramr @knobunc 
Review please.

**Comment by ramr** [comment]
LGTM   maybe add `source` to the white-list as well since we do use it + might make sense for tcp connections.

File: images/router/haproxy/conf/haproxy-config.template [hunk 1]
```diff
@@ -252,7 +252,13 @@ backend be_tcp_{{$cfgIdx}}
 {{ if ne (env "ROUTER_SYSLOG_ADDRESS" "") ""}}
   option tcplog
 {{ end }}
+    {{ with $balanceAlgo := index $cfg.Annotations "router.openshift.io/haproxy.balance" }}
+      {{ with $matchValue := (matchValues $balanceAlgo "roundrobin" "leastconn" ) }}
+  balance {{ $balanceAlgo }}
```
Comment (eparis) [inline images/router/haproxy/conf/haproxy-config.template]: I'm just template ignorant. What value does https://github.com/openshift/origin/pull/9710/files#diff-dc1a3cefa4a5177a9bfd475c81220cf1R256 bring? Does it cause the entire template processing to fail if the match fails?

Comment (rajatchopra) [inline images/router/haproxy/conf/haproxy-config.template]: No, the template processing does not fail. It just means that the 'with' block will not be processed further. Treat that as an 'if' statement.

**Comment by rajatchopra** [comment]
[test] again

**Comment by rajatchopra** [comment]
@eparis / @openshift/networking 
Merge ready?

**Comment by eparis** [comment]
[merge]

**Pull request merged**